syscall_test.go

```diff
@@ -77,14 +77,50 @@ func TestBindMountSuccessful(t *testing.T) {
 		t.Fatal(err)
 	}
 
-	err = bindMount(source, dest)
+	err = bindMount(source, dest, false)
 	if err != nil {
 		t.Fatal(err)
 	}
 
 	syscall.Unmount(dest, 0)
 }
 
+func TestBindMountReadonlySuccessful(t *testing.T) {
+	if os.Geteuid() != 0 {
+		t.Skip(testDisabledAsNonRoot)
+	}
+
+	source := filepath.Join(testDir, "fooDirSrc")
+	dest := filepath.Join(testDir, "fooDirDest")
+	syscall.Unmount(dest, 0)
+	os.Remove(source)
+	os.Remove(dest)
+
+	err := os.MkdirAll(source, mountPerm)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	err = os.MkdirAll(dest, mountPerm)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	err = bindMount(source, dest, true)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// should not be able to create file in read-only mount
+	destFile := filepath.Join(dest, "foo")
+	_, err = os.OpenFile(destFile, os.O_CREATE, mountPerm)
+	if err == nil {
+		t.Fatal(err)
+	}
+
+	syscall.Unmount(dest, 0)
```

Comment:
Should be called through a `defer` call after we do the `bindMount()`. We won't unmount in case the call to `os.OpenFile()` fails.